middlewares/authenticate.js

```diff
@@ -83,13 +84,31 @@ module.exports = async (req, res, next) => {
 
         // add user data to `req.userData` for further use
         req.userData = await users.fetchUser({ userId })
-
         return checkRestricted(req, res, next)
-      } else {
-        return res.boom.unauthorized('Unauthenticated User')
       }
-    } else {
-      return res.boom.unauthorized('Unauthenticated User')
     }
+
+    // If auth is not strictly required, just call the next handler.
+    if (!strict) {
+      return next()
+    }
```

Comment:
I feel this doesn't belong in this function. The `authenticate` function should ideally have the singular responsibility to just authenticate.
Whether to call it or not, should be the responsibility of the callsite.

Reply:
`authenticate` isn't just a function, it's a middleware. 

I can refactor the core auth logic into a separate function and call that optionally in the middleware based on flag.

Reply:
Done